### PR TITLE
Give understanable error when a null date is received

### PR DESCRIPTION
**Pull request by chris-moreton**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-4797

### Change description ###

When CCD sends a null date field, provide an error rather than allow null pointer exception to be thrown.

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/util/SscsOcrDataUtil.java [hunk 1]
```diff
@@ -80,10 +80,14 @@ public static String generateDateForCcd(Map<String, Object> pairs, List<String>
     public static String getDateForCcd(String ocrField, List<String> errors, String errorMessage) {
         DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy");
 
-        try {
-            return LocalDate.parse(ocrField, formatter).toString();
-        } catch (DateTimeParseException ex) {
+        if (ocrField == null) {
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/sscs/util/SscsOcrDataUtil.java]: You can leverage Apache Commons `StringUtils.isEmpty(str)` which checks for empty strings and handles null gracefully.

File: src/test/java/uk/gov/hmcts/reform/sscs/transformers/SscsCaseTransformerTest.java [hunk 2]
```diff
@@ -243,6 +243,15 @@ public void givenAnInvalidMrnDate_thenAddErrorToList() {
         assertTrue(result.getErrors().contains("mrn_date is an invalid date field. Needs to be in the format dd/mm/yyyy"));
     }
 
+    @Test
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/sscs/transformers/SscsCaseTransformerTest.java]: Also may be worth adding a test case for empty ?

File: src/main/java/uk/gov/hmcts/reform/sscs/util/SscsOcrDataUtil.java (hunk 1 above)
Comment (chris-moreton) [reply src/main/java/uk/gov/hmcts/reform/sscs/util/SscsOcrDataUtil.java]: Thanks, done.

File: src/test/java/uk/gov/hmcts/reform/sscs/transformers/SscsCaseTransformerTest.java (hunk 2 above)
Comment (chris-moreton) [reply src/test/java/uk/gov/hmcts/reform/sscs/transformers/SscsCaseTransformerTest.java]: Done
